src/hooks/useCourseItem.tsx

```diff
@@ -0,0 +1,22 @@
+import { useQuery, useMutation } from 'react-query';
+import { CourseItem } from 'api';
+import { AddItemData } from 'components/AddItemModal';
+
+import {
+	fetchGetCourseItem,
+	fetchPostCourseItem,
+	fetchPutCourseItem,
+	fetchDeleteCourseItem,
+} from 'api/courseItemOperations';
+
+export const useReadCourseItem = (id: string) =>
+	useQuery('courseItem', () => fetchGetCourseItem(id));
+
+export const useCreateCourseItem = () =>
+	useMutation((data: AddItemData) => fetchPostCourseItem(data));
+
+export const useUpdateCourseItem = () =>
+	useMutation((data: Partial<CourseItem>) => fetchPutCourseItem(data));
+
+export const useDeleteCourseItem = () =>
+	useMutation((id: string) => fetchDeleteCourseItem(id));
```

Comment:
@BraidenCutforth Due to the nature of the data and hooks altogether, I ended up making separate hooks for each operation, with slightly different react query hooks inside. Is this what you had in mind?

Reply:
I looked over this for a very long time and I think this is probably the best solution!